client/src/main/scala/org/apache/celeborn/client/commit/CommitHandler.scala

```diff
@@ -0,0 +1,365 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.celeborn.client.commit
+
+import java.util
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.atomic.{AtomicLong, LongAdder}
+
+import scala.collection.JavaConverters._
+
+import org.apache.celeborn.client.CommitManager.CommittedPartitionInfo
+import org.apache.celeborn.client.LifecycleManager.{ShuffleAllocatedWorkers, ShuffleFailedWorkers, ShuffleFileGroups, ShuffleMapperAttempts}
+import org.apache.celeborn.client.ShuffleCommittedInfo
+import org.apache.celeborn.common.CelebornConf
+import org.apache.celeborn.common.internal.Logging
+import org.apache.celeborn.common.meta.{PartitionLocationInfo, WorkerInfo}
+import org.apache.celeborn.common.protocol.{PartitionLocation, PartitionType}
+import org.apache.celeborn.common.protocol.message.ControlMessages.{CommitFiles, CommitFilesResponse}
+import org.apache.celeborn.common.protocol.message.StatusCode
+import org.apache.celeborn.common.rpc.RpcEndpointRef
+import org.apache.celeborn.common.util.{ThreadUtils, Utils}
+// Can Remove this if celeborn don't support scala211 in future
+import org.apache.celeborn.common.util.FunctionConverter._
+
+case class ParallelCommitResult(
+    masterPartitionLocationMap: ConcurrentHashMap[String, PartitionLocation],
+    slavePartitionLocationMap: ConcurrentHashMap[String, PartitionLocation],
+    commitFilesFailedWorkers: ShuffleFailedWorkers)
+
+abstract class CommitHandler(
+    appId: String,
+    conf: CelebornConf,
+    allocatedWorkers: ShuffleAllocatedWorkers,
+    reducerFileGroupsMap: ShuffleFileGroups,
+    committedPartitionInfo: CommittedPartitionInfo) extends Logging {
+
+  private val pushReplicateEnabled = conf.pushReplicateEnabled
+  private val testRetryCommitFiles = conf.testRetryCommitFiles
+  private val commitEpoch = new AtomicLong()
+  private val totalWritten = new LongAdder
+  private val fileCount = new LongAdder
+
+  def getPartitionType(): PartitionType
+
+  def isStageEnd(shuffleId: Int): Boolean = false
+
+  def isStageEndOrInProcess(shuffleId: Int): Boolean = false
```

Comment:
I think it is appropriate to encapsulate such a method, because it represents a state as a whole. If we use `isStageEnd || inProcess ` everywhere, it may be difficult to maintain.